include/interaction/Simulation.h

```diff
@@ -0,0 +1,89 @@
+#ifndef SIMULATION_H
+#define SIMULATION_H
```

Comment:
why is this one all caps and the others are CamelCase?

Reply:
The standard to define [Include guards](https://docs.microsoft.com/en-us/cpp/cpp/header-files-cpp?view=msvc-170) is to use the same name of the class in CAPS _H